## This data is giveing without quastions soo we need to make quastions and answer it

In [1]:
import pandas as pd
import numpy as np
# import matplotlib.pyplot as plt
# import seaborn as sns
import os

### but first of all lets see what the data looks like

In [2]:
databasse = pd.read_excel("C://Users//abdh-//Downloads//startup-expansion.xlsx")
databasse.head(5)

Store ID           City       State Sales Region New Expansion  \
0         1         Peoria     Arizona     Region 2           Old   
1         2        Midland       Texas     Region 2           Old   
2         3        Spokane  Washington     Region 2           Old   
3         4         Denton       Texas     Region 2           Old   
4         5  Overland Park      Kansas     Region 2           Old   

   Marketing Spend  Revenue  
0             2601    48610  
1             2727    45689  
2             2768    49554  
3             2759    38284  
4             2869    59887

By first look at the data we can say, this data for ('Revenue' , 'Marketing Spend' FOR the 'States', 'Cities' STORES) and(the new stores , and what region that store at)

so lets analysis the data

In [3]:
databasse.info() #main information for the data set

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Store ID         150 non-null    int64 
 1   City             150 non-null    object
 2   State            150 non-null    object
 3   Sales Region     150 non-null    object
 4   New Expansion    150 non-null    object
 5   Marketing Spend  150 non-null    int64 
 6   Revenue          150 non-null    int64 
dtypes: int64(3), object(4)
memory usage: 8.3+ KB


In [4]:
databasse.shape #the shape of the data

(150, 7)

In [5]:
databasse.nunique()

Store ID           150
City               149
State               35
Sales Region         2
New Expansion        2
Marketing Spend    142
Revenue            150
dtype: int64

In [6]:
databasse.isna().sum() #is there null?

Store ID           0
City               0
State              0
Sales Region       0
New Expansion      0
Marketing Spend    0
Revenue            0
dtype: int64

In [7]:
databasse.duplicated().sum() #is there duplicated?

0

In [8]:
databasse.describe().round(2)

Store ID  Marketing Spend   Revenue
count    150.00           150.00    150.00
mean      75.50          2893.15  39301.43
std       43.45           367.86  15465.75
min        1.00          1811.00  15562.00
25%       38.25          2662.25  21113.50
50%       75.50          2898.00  42993.00
75%      112.75          3111.50  51145.50
max      150.00          3984.00  68828.00

## by looking at the output of describe: we can see there is 150 STORES , and the MEAN of all 'Marketing Spend' = 2893.15 , and the MIN = 1811.00, and the MAX = 3984.00 ,,,,, and all the MEAN 'Revenue' = 39301.43 , and the MIN = 15562.00 , and the MAX = 68828.00

so lets EDA(Exploratory data analysis)

In [9]:
databasse.drop('Store ID',axis=1,inplace=True) #drop the store id bcz we dont need it

In [10]:
databasse[["State" , "Sales Region" , "New Expansion"]].value_counts()

State           Sales Region  New Expansion
California      Region 2      Old              37
Texas           Region 2      Old              15
Florida         Region 1      Old              12
Washington      Region 2      Old               7
Colorado        Region 2      Old               5
Alabama         Region 1      Old               4
Connecticut     Region 1      Old               4
Georgia         Region 1      Old               4
Kansas          Region 2      Old               3
Utah            Region 2      Old               3
Michigan        Region 1      Old               3
Massachusetts   Region 1      Old               3
New York        Region 1      Old               3
New Jersey      Region 1      Old               3
Iowa            Region 1      Old               3
Illinois        Region 1      Old               3
California      Region 2      New               3
South Carolina  Region 1      Old               3
North Carolina  Region 1      Old               2
Tennes

In [11]:
databasse.Revenue.sum()

5895215

In [12]:
databasse['Marketing Spend'].sum()

433972

In [13]:
databasse.State.value_counts() #the Stores in one State

California        40
Texas             17
Florida           12
Washington         7
Colorado           5
Illinois           5
Georgia            4
Alabama            4
Connecticut        4
New Jersey         4
Arizona            3
Tennessee          3
Iowa               3
Michigan           3
South Carolina     3
Utah               3
Massachusetts      3
Kansas             3
New York           3
Louisiana          2
North Carolina     2
Ohio               2
Virginia           2
Oregon             2
Mississippi        1
New Mexico         1
Arkansas           1
New Hampshire      1
Oklahoma           1
Montana            1
Indiana            1
South Dakota       1
Minnesota          1
Wisconsin          1
Rhode Island       1
Name: State, dtype: int64

so the most stores in 1 state is in California = 40 stores , Texas = 17 , Florida = 12 , Washington = 7

In [14]:
databasse.City.value_counts() #the Stores in one City

Rochester        2
Killeen          1
Wichita Falls    1
Naperville       1
Clovis           1
                ..
Akron            1
Fullerton        1
Manchester       1
Everett          1
Thousand Oaks    1
Name: City, Length: 149, dtype: int64

so the most stores in 1 citys is in Rochester = 2 stores , and the rest is 1

In [15]:
databasse['Sales Region'].value_counts() #the Stores in one Region

Region 2    86
Region 1    64
Name: Sales Region, dtype: int64

In [16]:
databasse['New Expansion'].value_counts() #the stores in one Expansion

Old    140
New     10
Name: New Expansion, dtype: int64

In [17]:
databasse.groupby('Sales Region')['Revenue'].describe().round(2)

count      mean       std      min       25%      50%       75%  \
Sales Region                                                                    
Region 1       64.0  40567.20  15041.26  17006.0  22717.25  45217.5  52082.50   
Region 2       86.0  38359.47  15795.49  15562.0  20677.50  41235.5  50138.75   

                  max  
Sales Region           
Region 1      65475.0  
Region 2      68828.0

So the MEAN in Region_1 is more (40567.20) than the Region_2 (38359.47)
and the MIN in Region_1 is more (17006.0) than the Region_2 (15562.0)
but the MAX in Region_1 is less (65475.0) than the Region_2 (68828.0)
## so the Conclusion of this is : the best Region in Revenue is the  Region who get more revenue = Region_1(MEAN)

In [18]:
databasse.groupby('Sales Region')['Marketing Spend'].describe().round(2)

count     mean     std     min      25%     50%      75%     max
Sales Region                                                                  
Region 1       64.0  2889.00  322.55  2028.0  2682.75  2893.5  3111.25  3586.0
Region 2       86.0  2896.23  400.10  1811.0  2644.25  2898.0  3105.50  3984.0

So the MEAN in Region_1 is less (2889.00) than the Region_2 (2896.23)
and the MIN in Region_1 is more (2028.0) than the Region_2 (1811.0)
but the MAX in Region_1 is less (3586.0) than the Region_2 (3984.0)
## so the Conclusion of this is : the best Region in Marketing Spend is the region who spend lees = Region_1(mean)

In [19]:
databasse.groupby('Sales Region')['Revenue'].sum()

Sales Region
Region 1    2596301
Region 2    3298914
Name: Revenue, dtype: int64

## the Revenue in Region_1(2596301) is less than Region_2(3298914)
## so the most Revenue is coming from Region_2

In [20]:
databasse.groupby('Sales Region')['Marketing Spend'].sum()

Sales Region
Region 1    184896
Region 2    249076
Name: Marketing Spend, dtype: int64

## the Marketing Spend in Region_1(184896) is less than Region_2(249076)
## so the less Marketing Spend is Region_1

In [21]:
databasse.groupby('New Expansion')['Revenue'].describe().round(2)

count      mean       std      min      25%      50%      75%  \
New Expansion                                                                  
New             10.0  45809.40  11414.54  22457.0  40950.0  48634.5  51618.5   
Old            140.0  38836.58  15642.78  15562.0  20694.5  41743.0  50978.5   

                   max  
New Expansion           
New            63148.0  
Old            68828.0

So the MEAN in New is more (45809.40) than the Old (38836.58)
and the MIN in New is more (22457.0) than the Old (15562.0)
but the MAX in New is less (63148.0) than the Old (68828.0)
## so the Conclusion of this is : the best Expansion in Revenue are who's get more Revenue = (New)Expansion(mean) 
but dont forgate The new store is just = 10 / and the old = 140

In [22]:
databasse.groupby('New Expansion')['Marketing Spend'].describe().round(2)

count     mean     std     min      25%     50%      75%  \
New Expansion                                                             
New             10.0  2908.40  496.60  2251.0  2485.25  2928.0  3207.75   
Old            140.0  2892.06  359.27  1811.0  2675.75  2894.5  3110.50   

                  max  
New Expansion          
New            3675.0  
Old            3984.0

So the MEAN in New is more (2908.40) than the Old (2892.06)
and the MIN in New is more (2251.0) than the Old (1811.0)
but the MAX in New is less (3675.0) than the Old (3984.0)
## so the Conclusion of this is : the best Expansion in Marketing Spend are who's get less Marketing Spend = (old)Expansion(mean) 
but dont forgate The new store is just = 10 / and the old = 140

In [23]:
databasse.groupby('State')['Revenue'].describe().round(2).sort_values(by= 'mean' , ascending=False).head(10)
#this groupby will show you The mean , min ,max ,how many store in one state//

count      mean       std      min       25%      50%       75%  \
State                                                                           
Arkansas        1.0  68828.00       NaN  68828.0  68828.00  68828.0  68828.00   
Rhode Island    1.0  62337.00       NaN  62337.0  62337.00  62337.0  62337.00   
Alabama         4.0  55256.25   5400.89  49290.0  51408.00  55698.5  59546.75   
Mississippi     1.0  54701.00       NaN  54701.0  54701.00  54701.0  54701.00   
New York        3.0  53348.67  11803.77  41313.0  47570.00  53827.0  59366.50   
Tennessee       3.0  50802.00   8901.32  40545.0  47951.00  55357.0  55930.50   
South Dakota    1.0  50583.00       NaN  50583.0  50583.00  50583.0  50583.00   
Michigan        3.0  49253.00  15323.82  35022.0  41142.00  47262.0  56368.50   
Virginia        2.0  48203.50  13324.01  38782.0  43492.75  48203.5  52914.25   
Louisiana       2.0  48197.50  11232.39  40255.0  44226.25  48197.5  52168.75   

                  max  
State                  
Arkansas      68828.0  
Rhode Island  62337.0  
Alabama       60338.0  
Mississippi   54701.0  
New York      64906.0  
Tennessee     56504.0  
South Dakota  50583.0  
Michigan      65475.0  
Virginia      57625.0  
Louisiana     56140.0

these are the best 10 state by mean 'Revenue'

In [24]:
databasse.groupby('State')['Marketing Spend'].describe().round(2).sort_values(by= 'mean' , ascending=False).head(10)

count     mean     std     min      25%     50%     75%  \
State                                                                     
Wisconsin         1.0  3586.00     NaN  3586.0  3586.00  3586.0  3586.0   
Mississippi       1.0  3466.00     NaN  3466.0  3466.00  3466.0  3466.0   
Rhode Island      1.0  3191.00     NaN  3191.0  3191.00  3191.0  3191.0   
Tennessee         3.0  3150.00  408.77  2777.0  2931.50  3086.0  3336.5   
Alabama           4.0  3142.50  224.88  2838.0  3042.00  3198.5  3299.0   
North Carolina    2.0  3141.00  171.12  3020.0  3080.50  3141.0  3201.5   
Colorado          5.0  3125.40  395.02  2642.0  2928.00  3083.0  3277.0   
Connecticut       4.0  3092.25  233.14  2901.0  2910.75  3038.0  3219.5   
Montana           1.0  3015.00     NaN  3015.0  3015.00  3015.0  3015.0   
Indiana           1.0  3005.00     NaN  3005.0  3005.00  3005.0  3005.0   

                   max  
State                   
Wisconsin       3586.0  
Mississippi     3466.0  
Rhode Island    3191.0  
Tennessee       3587.0  
Alabama         3335.0  
North Carolina  3262.0  
Colorado        3697.0  
Connecticut     3392.0  
Montana         3015.0  
Indiana         3005.0

these are the worst top 10 state by mean 'Marketing Spend'

In [25]:
databasse.groupby('City')['Revenue'].describe().round(2).sort_values(by= 'mean' , ascending=False).head(10)

count     mean  std      min      25%      50%      75%  \
City                                                                      
Little Rock       1.0  68828.0  NaN  68828.0  68828.0  68828.0  68828.0   
Grand Rapids      1.0  65475.0  NaN  65475.0  65475.0  65475.0  65475.0   
Oxnard            1.0  64302.0  NaN  64302.0  64302.0  64302.0  64302.0   
Brownsville       1.0  63148.0  NaN  63148.0  63148.0  63148.0  63148.0   
Fontana           1.0  63027.0  NaN  63027.0  63027.0  63027.0  63027.0   
Providence        1.0  62337.0  NaN  62337.0  62337.0  62337.0  62337.0   
Birmingham        1.0  60338.0  NaN  60338.0  60338.0  60338.0  60338.0   
Overland Park     1.0  59887.0  NaN  59887.0  59887.0  59887.0  59887.0   
San Bernardino    1.0  59870.0  NaN  59870.0  59870.0  59870.0  59870.0   
Worcester         1.0  59840.0  NaN  59840.0  59840.0  59840.0  59840.0   

                    max  
City                     
Little Rock     68828.0  
Grand Rapids    65475.0  
Oxnard          64302.0  
Brownsville     63148.0  
Fontana         63027.0  
Providence      62337.0  
Birmingham      60338.0  
Overland Park   59887.0  
San Bernardino  59870.0  
Worcester       59840.0

these are the best 10 citys by mean 'Revenue'

In [26]:
databasse.groupby('City')['Marketing Spend'].describe().round(2).sort_values(by= 'mean' , ascending=False).head(10)

count    mean  std     min     25%     50%     75%     max
City                                                                      
Amarillo          1.0  3984.0  NaN  3984.0  3984.0  3984.0  3984.0  3984.0
Richmond          1.0  3924.0  NaN  3924.0  3924.0  3924.0  3924.0  3924.0
Centennial        1.0  3697.0  NaN  3697.0  3697.0  3697.0  3697.0  3697.0
Burbank           1.0  3679.0  NaN  3679.0  3679.0  3679.0  3679.0  3679.0
Brownsville       1.0  3675.0  NaN  3675.0  3675.0  3675.0  3675.0  3675.0
Salt Lake City    1.0  3620.0  NaN  3620.0  3620.0  3620.0  3620.0  3620.0
Chattanooga       1.0  3587.0  NaN  3587.0  3587.0  3587.0  3587.0  3587.0
Green Bay         1.0  3586.0  NaN  3586.0  3586.0  3586.0  3586.0  3586.0
Moreno Valley     1.0  3507.0  NaN  3507.0  3507.0  3507.0  3507.0  3507.0
Miramar           1.0  3488.0  NaN  3488.0  3488.0  3488.0  3488.0  3488.0

these are the worst top 10 citys by mean 'Marketing Spend'

____________________________________________

## City, state : with Highest Revenue

In [27]:
databasse[databasse['Revenue']==databasse['Revenue'].max()][['City','State','Marketing Spend','Revenue']] #City with Highest Revenue

City     State  Marketing Spend  Revenue
63  Little Rock  Arkansas             2440    68828

## City, state: with Lowest Revenue

In [28]:
databasse[databasse['Revenue']==databasse['Revenue'].min()][['City','State','Marketing Spend','Revenue']] #City with Lowest Revenue

City       State  Marketing Spend  Revenue
53  Everett  Washington             3129    15562

## City , state with : Highest Marketing Spend

In [29]:
databasse[databasse['Marketing Spend']==databasse['Marketing Spend'].max()][['City','State','Marketing Spend','Revenue']] #City with Highest Marketing Spend

City  State  Marketing Spend  Revenue
86  Amarillo  Texas             3984    56089

## City , state : with Lowest Marketing Spend

In [30]:
databasse[databasse['Marketing Spend']==databasse['Marketing Spend'].min()][['City','State','Marketing Spend','Revenue']] #City with Lowest Marketing Spend

City       State  Marketing Spend  Revenue
135  San Mateo  California             1811    19426

## Largest Revenue in all data

In [31]:
databasse.nlargest(1 , columns='Revenue')

City     State Sales Region New Expansion  Marketing Spend  Revenue
63  Little Rock  Arkansas     Region 2           Old             2440    68828

## Smallest Revenue in all data

In [32]:
databasse.nsmallest(1, columns='Revenue')

City       State Sales Region New Expansion  Marketing Spend  Revenue
53  Everett  Washington     Region 2           Old             3129    15562

## Largest Revenue in all Region_1


In [33]:
df_reg_1 = databasse[databasse['Sales Region'] == 'Region 1']
df_reg_1.nlargest(1, columns='Revenue')

City     State Sales Region New Expansion  Marketing Spend  \
31  Grand Rapids  Michigan     Region 1           Old             2808   

    Revenue  
31    65475

# Largest Revenue in all Region_2

In [34]:
df_reg_2 = databasse[databasse['Sales Region'] == 'Region 2']
df_reg_2.nlargest(1, columns='Revenue')

City     State Sales Region New Expansion  Marketing Spend  Revenue
63  Little Rock  Arkansas     Region 2           Old             2440    68828

## Smallest Revenue in all Region_1

In [35]:
df_reg_1.nsmallest(1, columns='Revenue')

City          State Sales Region New Expansion  Marketing Spend  \
52  Manchester  New Hampshire     Region 1           Old             2374   

    Revenue  
52    17006

## Smallest Revenue in all Region_2

In [36]:
df_reg_2.nsmallest(1, columns='Revenue')

City       State Sales Region New Expansion  Marketing Spend  Revenue
53  Everett  Washington     Region 2           Old             3129    15562

## Smallest Marketing Spend in all data

In [37]:
databasse[databasse['Marketing Spend'] == min(databasse['Marketing Spend'])]

City       State Sales Region New Expansion  Marketing Spend  \
135  San Mateo  California     Region 2           Old             1811   

     Revenue  
135    19426

# Smallest Marketing Spend in Region_1

In [38]:
df_reg_1.nsmallest(1, columns='Marketing Spend')

City    State Sales Region New Expansion  Marketing Spend  Revenue
130  Savannah  Georgia     Region 1           Old             2028    36029

## Smallest Marketing Spend in Region_2

In [39]:
df_reg_2.nsmallest(1, columns='Marketing Spend')

City       State Sales Region New Expansion  Marketing Spend  \
135  San Mateo  California     Region 2           Old             1811   

     Revenue  
135    19426

## Largest Marketing Spend in data

In [40]:
databasse.nlargest(1, columns='Marketing Spend')

City  State Sales Region New Expansion  Marketing Spend  Revenue
86  Amarillo  Texas     Region 2           Old             3984    56089

## Largest Marketing Spend in all Region_1

In [41]:
df_reg_1.nlargest(1, columns='Marketing Spend')

City      State Sales Region New Expansion  Marketing Spend  Revenue
133  Green Bay  Wisconsin     Region 1           Old             3586    18454

## Largest Marketing Spend in all Region_2

In [42]:
df_reg_2.nlargest(1, columns='Marketing Spend')

City  State Sales Region New Expansion  Marketing Spend  Revenue
86  Amarillo  Texas     Region 2           Old             3984    56089

## Cities with low marketing spend than average and high revenue than average

In [43]:
best_city = databasse[(databasse['Marketing Spend'] < databasse['Marketing Spend'].mean()) & (databasse['Revenue'] > databasse['Revenue'].mean())]
best_city

City           State Sales Region New Expansion  \
0              Peoria         Arizona     Region 2           Old   
1             Midland           Texas     Region 2           Old   
2             Spokane      Washington     Region 2           Old   
4       Overland Park          Kansas     Region 2           Old   
8           Worcester   Massachusetts     Region 1           Old   
16        Tallahassee         Florida     Region 1           Old   
25     Port St. Lucie         Florida     Region 1           Old   
26          Elizabeth      New Jersey     Region 1           Old   
28               Waco           Texas     Region 2           Old   
31       Grand Rapids        Michigan     Region 1           Old   
37             Oxnard      California     Region 2           Old   
39         Cape Coral         Florida     Region 1           Old   
40           Syracuse        New York     Region 1           Old   
41       Newport News        Virginia     Region 1           Old   
45           Pasadena      California     Region 2           Old   
50              Akron            Ohio     Region 1           Old   
55           Thornton        Colorado     Region 2           Old   
60           Columbia  South Carolina     Region 1           Old   
61        Simi Valley      California     Region 2           Old   
63        Little Rock        Arkansas     Region 2           Old   
73       Murfreesboro       Tennessee     Region 2           Old   
75             Topeka          Kansas     Region 2           Old   
76   West Valley City            Utah     Region 2           Old   
79         Costa Mesa      California     Region 2           Old   
89             Tacoma      Washington     Region 2           Old   
91            Augusta         Georgia     Region 1           Old   
93             Aurora        Illinois     Region 1           Old   
95             Dayton            Ohio     Region 1           Old   
97         Naperville        Illinois     Region 1           Old   
102           Visalia      California     Region 2           Old   
113            Renton      Washington     Region 2           Old   
114           McAllen           Texas     Region 2           Old   
117          Bellevue      Washington     Region 2           Old   
118  Huntington Beach      California     Region 2           Old   
121            Mobile         Alabama     Region 1           Old   
126      Cedar Rapids            Iowa     Region 1           Old   
127          Columbus         Georgia     Region 1           Old   
136            Warren        Michigan     Region 1           Old   
138         Lafayette       Louisiana     Region 1           Old   
144          Glendale      California     Region 2           New   
147          Rockford        Illinois     Region 1           New   
149     Thousand Oaks      California     Region 2           New   

     Marketing Spend  Revenue  
0               2601    48610  
1               2727    45689  
2               2768    49554  
4               2869    59887  
8               2675    59840  
16              2737    47729  
25              2420    55790  
26              2557    45017  
28              2483    39744  
31              2808    65475  
37              2318    64302  
39              2886    52250  
40              2373    41313  
41              2758    57625  
45              2790    48254  
50              2338    50364  
55              2642    46490  
60              2874    42803  
61              2792    41110  
63              2440    68828  
73              2777    40545  
75              2854    40921  
76              2555    49609  
79              2275    41361  
89              2552    45666  
91              2676    47482  
93              2685    50650  
95              2805    42026  
97              2699    39856  
102             2104    45632  
113             2754    44635  
114             2282    43924  
117             2542    40462  
118 

In [44]:
top_suc_state_city = best_city.groupby('State', as_index=False ).count().sort_values( by='Revenue' , ascending=False ).reset_index()[['State' , 'City']]
top_suc_state_city

State  City
0       California     8
1       Washington     4
2          Florida     3
3         Illinois     3
4            Texas     3
5          Georgia     2
6           Kansas     2
7         Michigan     2
8             Ohio     2
9         New York     1
10        Virginia     1
11            Utah     1
12       Tennessee     1
13  South Carolina     1
14         Alabama     1
15      New Jersey     1
16         Arizona     1
17       Louisiana     1
18            Iowa     1
19        Colorado     1
20        Arkansas     1
21   Massachusetts     1

# By this data we can say that California is most suitable city for most successfull startups with low spend,followed by Washington

______________________________________

lets add the Profit and ROMS

profit Formula is (Profit = Revenue - Marketing Spend)

In [45]:
databasse['profit'] = databasse['Revenue'] - databasse["Marketing Spend"]
databasse

City       State Sales Region New Expansion  Marketing Spend  \
0             Peoria     Arizona     Region 2           Old             2601   
1            Midland       Texas     Region 2           Old             2727   
2            Spokane  Washington     Region 2           Old             2768   
3             Denton       Texas     Region 2           Old             2759   
4      Overland Park      Kansas     Region 2           Old             2869   
..               ...         ...          ...           ...              ...   
145         Paterson  New Jersey     Region 1           New             2251   
146      Brownsville       Texas     Region 2           New             3675   
147         Rockford    Illinois     Region 1           New             2648   
148  College Station       Texas     Region 2           New             2994   
149    Thousand Oaks  California     Region 2           New             2431   

     Revenue  profit  
0      48610   46009  
1      45689   42962  
2      49554   46786  
3      38284   35525  
4      59887   57018  
..       ...     ...  
145    34603   32352  
146    63148   59473  
147    43377   40729  
148    22457   19463  
149    40141   37710  

[150 rows x 7 columns]

In [46]:
databasse.sample(4)

City       State Sales Region New Expansion  Marketing Spend  \
135     San Mateo  California     Region 2           Old             1811   
78         Pueblo    Colorado     Region 2           Old             2928   
31   Grand Rapids    Michigan     Region 1           Old             2808   
6      Birmingham     Alabama     Region 1           Old             3110   

     Revenue  profit  
135    19426   17615  
78     19563   16635  
31     65475   62667  
6      60338   57228

ROMS Formula is (ROMS = (profit / Marketing Spend)* 100)

In [47]:
databasse['ROMS'] = round((databasse['profit'] / databasse["Marketing Spend"]) * 100, 2)

In [48]:
databasse['ROMS'].round(2)

0      1768.90
1      1575.43
2      1690.25
3      1287.60
4      1987.38
        ...   
145    1437.23
146    1618.31
147    1538.10
148     650.07
149    1551.21
Name: ROMS, Length: 150, dtype: float64

ROMS% Formula is (ROMS% = (ROMS%) / 100 , 2)

In [49]:
databasse['ROMS%'] = round(databasse['ROMS'] / 100 , 2)
databasse.head(5)

City       State Sales Region New Expansion  Marketing Spend  \
0         Peoria     Arizona     Region 2           Old             2601   
1        Midland       Texas     Region 2           Old             2727   
2        Spokane  Washington     Region 2           Old             2768   
3         Denton       Texas     Region 2           Old             2759   
4  Overland Park      Kansas     Region 2           Old             2869   

   Revenue  profit     ROMS  ROMS%  
0    48610   46009  1768.90  17.69  
1    45689   42962  1575.43  15.75  
2    49554   46786  1690.25  16.90  
3    38284   35525  1287.60  12.88  
4    59887   57018  1987.38  19.87

In [50]:
databasse

City       State Sales Region New Expansion  Marketing Spend  \
0             Peoria     Arizona     Region 2           Old             2601   
1            Midland       Texas     Region 2           Old             2727   
2            Spokane  Washington     Region 2           Old             2768   
3             Denton       Texas     Region 2           Old             2759   
4      Overland Park      Kansas     Region 2           Old             2869   
..               ...         ...          ...           ...              ...   
145         Paterson  New Jersey     Region 1           New             2251   
146      Brownsville       Texas     Region 2           New             3675   
147         Rockford    Illinois     Region 1           New             2648   
148  College Station       Texas     Region 2           New             2994   
149    Thousand Oaks  California     Region 2           New             2431   

     Revenue  profit     ROMS  ROMS%  
0      48610   46009  1768.90  17.69  
1      45689   42962  1575.43  15.75  
2      49554   46786  1690.25  16.90  
3      38284   35525  1287.60  12.88  
4      59887   57018  1987.38  19.87  
..       ...     ...      ...    ...  
145    34603   32352  1437.23  14.37  
146    63148   59473  1618.31  16.18  
147    43377   40729  1538.10  15.38  
148    22457   19463   650.07   6.50  
149    40141   37710  1551.21  15.51  

[150 rows x 9 columns]

## so we came to end of this analysis, There a lot of information in this data so ill try to Build a dashbourd from it

In [51]:
# databasse.to_csv("C://Users//abdh-//OneDrive/سطح المكتب//projictt1.csv") ##print the data on csv file

#### So we have 150 stores, in 35 state and 2 salse Region, some of it is new stores : 7% , and the old is 93% of the store , and the Revenue of all stores is : 5895215 , and all the markiting is : 433972, and we have 86 stores in Region_2 , and 64 stores in Region_1, 140 is old , and the new ones is 10, and the most mony gose to the Region_2, and also is the most spending, .............  .

Thank you